src/app/campaigns/campaign-edit/campaign-edit.component.ts

```diff
@@ -98,4 +106,15 @@ export class CampaignEditComponent {
     this.updateForm.value.documents.splice(index, 1)
     this.updateForm.markAsDirty()
   }
+
+  addNewsUrl() {
+    this.newsUrls.push(this.fb.group({
+      url: ['', Validators.required]
+    }))
```

Comment:
Why adding a group on every new link? It is that how it should be done?

Reply:
I found examples for form arrays which only had multiple fields, so they had to use form groups for each element. I am not sure how to bind individual form controls here, but I will have a look at the course on Udemy, I think there was an example there how to do it.